nx-os/poap/poap.py

```diff
@@ -72,13 +73,10 @@
 # parameters passed through environment:
 # TODO: use good old argv[] instead, using env is bad idea.
 # pid is used for temp file name: just use getpid() instead!
-# serial number should be gotten from "show version" or something!
 pid=""
 if os.environ.has_key('POAP_PID'):
     pid=os.environ['POAP_PID']
-serial_number=None
-if os.environ.has_key('POAP_SERIAL'):
-    serial_number=os.environ['POAP_SERIAL']
+serial_number=re.search("ID\ (.+?)\n", cli("show version | grep Board")).group(1)
```

Comment:
Now that I've had a chance to actually try this on some N9K's, this likely won't do what we want across the N9K product line.  For fixed configuration switches (leafs) it seems fine but for modular switches this will return the serial number of the supervisor and not the serial number of the chassis.  Since the supervisor can be swapped on a chassis we should be using that instead.  So combining my previous suggestion with this change would be:

    serial_number = cli('show inventory chassis | grep SN:').rstrip().split(" ")[-1]

Please do use spaces around the = (we should do the same for the rest of the file).  I'm slowly trying to get this to be pep8 compliant or something closer to it.

Thanks!